### ISSUE TITLE
RangeToolLink doesn't work with overlays

### ISSUE THREAD

**Issue by ahuang11** [holoviz/holoviews#4472]
```
import holoviews as hv
from holoviews import opts
from bokeh.sampledata.stocks import AAPL

aapl_df = pd.DataFrame({'open': AAPL['open'], 'close': AAPL['close']}, index=pd.to_datetime(AAPL['date']))
aapl_df.index.name = 'Date'

aapl_curve = aapl_df.hvplot('Date', 'close')
tgt = aapl_curve.relabel('AAPL close price').opts('Overlay', width=800, labelled=['y'], toolbar='disable')
src = aapl_curve.opts('Overlay', width=800, height=100, yaxis=None, default_tools=[])

RangeToolLink(src, tgt)

layout = (tgt + src).cols(1)
layout.opts(opts.Layout(shared_axes=False, merge_tools=False))
```

expected a tool to show up in the toolbar like that:
http://holoviews.org/gallery/demos/bokeh/timeseries_range_tool.html

**Comment by droumis** [comment]
Maybe I'm misunderstanding, but the comment does not include an overlay. 

Here is an overlay version and workaround to get RangeToolLink working... You have to subselect a single element in the target plot rather than the overlay

```
import holoviews as hv
from holoviews import opts
from bokeh.sampledata.stocks import AAPL
import pandas as pd
import hvplot.pandas

aapl_df = pd.DataFrame({'open': AAPL['open'], 'close': AAPL['close']}, index=pd.to_datetime(AAPL['date']))
aapl_df.index.name = 'Date'

aapl_curve_close = aapl_df.hvplot('Date', 'close')
aapl_curve_open = aapl_df.hvplot('Date', 'open')
tgt = aapl_curve_close * aapl_curve_open

aapl_df['change'] = aapl_df['close'] - aapl_df['open']
src = aapl_df.hvplot('Date','change')

RangeToolLink(src, list(tgt.values())[0])


layout = (tgt + src).cols(1)
layout.opts(opts.Layout(shared_axes=False, merge_tools=False))
```
![image](https://github.com/holoviz/holoviews/assets/6613202/c95f6f06-fc94-4da7-b33f-bec6871e84ce)

**Comment by droumis** [comment]
I'm adding this to the CZI project list, as I'd like the API to be improved to work with overlays without the workaround

**Comment by droumis** [comment]
closed by #5881
